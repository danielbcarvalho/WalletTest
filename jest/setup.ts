import 'react-native-gesture-handler/jestSetup';
import { mockDefaultTheme } from '../src/theme/defaultTheme';

jest.mock('styled-components', () => {
  const original = jest.requireActual('styled-components');

  return {
    ...original,
    useTheme: () => {
      return mockDefaultTheme;
    },
  };
});
