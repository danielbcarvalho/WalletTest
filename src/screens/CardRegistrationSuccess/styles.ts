import styled from 'styled-components/native';

export const Title = styled.Text`
  font-size: ${({ theme }) => theme.typography.h4.fontSize};
  color: ${({ theme }) => theme.pallete.textOnBackground.default};
`;

export const CardWrapper = styled.View`
  width: 100%;
  margin: 30px 0 20px 0;
`;
