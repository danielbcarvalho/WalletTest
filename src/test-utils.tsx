// TODO: Verificar essa melhoria nos testes
import React from 'react';
import {
  render,
  RenderAPI,
  RenderOptions,
} from '@testing-library/react-native';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';

import { mockDefaultTheme } from './theme/defaultTheme';

interface IAllThemeProviders {
  children: React.ReactElement;
}
const AllTheProviders = ({ children }: IAllThemeProviders) => {
  return <ThemeProvider theme={mockDefaultTheme}>{children}</ThemeProvider>;
};

function customRender(
  ui: React.ReactElement,
  options = {} as RenderOptions,
): RenderAPI {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

export function rendererCreate(
  component: React.ReactElement,
): ReactTestRenderer {
  return renderer.create(<AllTheProviders>{component}</AllTheProviders>);
}

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
