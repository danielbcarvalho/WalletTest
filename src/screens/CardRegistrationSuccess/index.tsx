import * as React from 'react';

import { useTranslation } from 'react-i18next';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';

import Card from '../../components/Card';
import { ParamsList } from '../../Routes';
import BrandTitle from '../../components/Title';
import { Button } from '../../components/Button';
import StyledBackgroundContainer from '../../components/StyledBackgoundContainer';

import { Title, CardWrapper } from './styles';

type CardRegistrationSuccessRouteProp = RouteProp<
  ParamsList,
  'CardRegistrationSuccess'
>;

type CardRegistrationSuccessScreenProp = NativeStackNavigationProp<
  ParamsList,
  'CardRegistrationSuccess'
>;

function CardRegistrationSuccess() {
  const { t } = useTranslation();
  const { params } = useRoute<CardRegistrationSuccessRouteProp>();
  const { navigate } = useNavigation<CardRegistrationSuccessScreenProp>();

  function handleAdvance() {
    navigate('CardList');
  }

  return (
    <StyledBackgroundContainer>
      <BrandTitle />
      <Title>{t('successfully registered card')}</Title>
      <CardWrapper>
        <Card card={params.card} />
      </CardWrapper>
      <Button onPress={handleAdvance} title={t('advance')} />
    </StyledBackgroundContainer>
  );
}

export default CardRegistrationSuccess;
