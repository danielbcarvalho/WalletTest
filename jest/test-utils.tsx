import React, { ReactElement, FC } from 'react';
import { render } from '@testing-library/react-native';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ThemeProvider } from '../src/theme';

type Options = Parameters<typeof render>[1];

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      cacheTime: Infinity,
    },
  },
});
const AllTheProviders: FC<{ children: React.ReactNode }> = ({ children }) => (
  <ThemeProvider>
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  </ThemeProvider>
);

const customRender = (ui: ReactElement, options?: Options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
