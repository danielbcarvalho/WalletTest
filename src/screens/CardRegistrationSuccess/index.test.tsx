import React from 'react';
import {
  NavigationContainer,
  useNavigation,
  useRoute,
} from '@react-navigation/native';
import { fireEvent, render } from '../../../jest/test-utils';
import CardRegistrationSuccess from './';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: jest.fn(),
  }),
}));

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
  useRoute: jest.fn(),
}));

describe('CardRegistrationSuccess', () => {
  let navigateMock: jest.Mock;

  beforeEach(() => {
    navigateMock = jest.fn();
    useNavigation.mockReturnValue({
      navigate: navigateMock,
    });
    useRoute.mockReturnValue({
      params: {
        card: {
          id: '4ec42ba9-50af-40d2-af90-8312edbd9ca2',
          number: '3529 5435 3355 8720',
          cvv: '017',
          name: 'John Doe',
          expiry: '12/23',
          kind: 'black',
        },
      },
    });
  });

  const component = (
    <NavigationContainer>
      <CardRegistrationSuccess />
    </NavigationContainer>
  );

  test('navigate to CardList on button press', () => {
    const { getByTestId } = render(component);
    fireEvent.press(getByTestId('multipurpose-button'));
    expect(navigateMock).toHaveBeenCalledWith('CardList');
  });

  test('render correctly', () => {
    const { toJSON } = render(component);
    expect(toJSON()).toMatchSnapshot();
  });
});
