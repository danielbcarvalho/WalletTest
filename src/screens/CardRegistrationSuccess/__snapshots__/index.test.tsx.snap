// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardRegistrationSuccess render correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#142995",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingLeft": 30,
      "paddingRight": 30,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#EEEEEE",
        "borderBottomLeftRadius": 50,
        "borderBottomRightRadius": 50,
        "height": 230,
        "opacity": 0.2,
        "position": "absolute",
        "right": 120,
        "top": -70,
        "transform": [
          {
            "rotate": "-39.93deg",
          },
        ],
        "width": 370,
      }
    }
    top={true}
  />
  <View
    style={
      {
        "backgroundColor": "#EEEEEE",
        "borderBottomLeftRadius": 50,
        "borderBottomRightRadius": 50,
        "bottom": -70,
        "height": 230,
        "left": 120,
        "opacity": 0.2,
        "position": "absolute",
        "transform": [
          {
            "rotate": "139deg",
          },
        ],
        "width": 370,
      }
    }
  />
  <Text
    style={
      {
        "color": "#FFF",
        "fontFamily": "PTSans-Caption",
        "fontSize": 28,
        "lineHeight": 52,
      }
    }
  />
  <Text
    style={
      {
        "color": "#FFF",
        "fontSize": 20,
      }
    }
  />
  <View
    style={
      {
        "marginBottom": 20,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 30,
        "width": "100%",
      }
    }
  >
    <View
      kind="black"
      style={
        {
          "alignItems": "flex-start",
          "backgroundColor": "#000000",
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "display": "flex",
          "gap": 10,
          "height": 180,
          "justifyContent": "space-around",
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
          "width": "100%",
        }
      }
    >
      <Text
        kind="black"
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "PTSans-Regular",
            "fontSize": 18,
            "lineHeight": 20,
          }
        }
      >
        Black Card
      </Text>
      <View
        style={
          {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <Text
          kind="black"
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "PTSans-Regular",
              "fontSize": 16,
              "lineHeight": 18,
            }
          }
        >
          John Doe
        </Text>
        <Text
          kind="black"
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "PTSans-Regular",
              "fontSize": 14,
              "lineHeight": 16,
              "textAlign": "center",
            }
          }
        >
          **** **** **** 8720
        </Text>
        <Text
          kind="black"
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "PTSans-Regular",
              "fontSize": 14,
              "lineHeight": 16,
              "textAlign": "center",
            }
          }
        >
          undefined 12/23
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#12C2E9",
        "borderBottomLeftRadius": 12,
        "borderBottomRightRadius": 12,
        "borderTopLeftRadius": 12,
        "borderTopRightRadius": 12,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 10,
        "maxHeight": 56,
        "minHeight": 56,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
    testID="multipurpose-button"
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "PTSans-Regular",
          "fontSize": 18,
        }
      }
      type="primary"
    />
  </View>
</RNCSafeAreaView>
`;
